test/GroupByTest.cpp

```diff
@@ -1912,3 +1911,282 @@ TEST(GroupBy, Descriptor) {
+          << "Not enough idTables yielded. Expected: " << idTables.size();
+    } else {
+      IdTable aggregatedTable{groupBy.getResultWidth(), groupBy.allocator()};
+      for (const IdTable& idTable : idTables) {
+        aggregatedTable.insertAtEnd(idTable);
+      }
+      EXPECT_EQ(result.idTable(), aggregatedTable);
+    }
+  }
+};
+}  // namespace
+
+INSTANTIATE_TEST_SUITE_P(FalseTrue, GroupByLazyFixture, testing::Bool());
+
+// _____________________________________________________________________________
+TEST_P(GroupByLazyFixture, testEmptyGeneratorYieldsEmptyResult) {
+  auto subtree = makeExecutionTree<ValuesForTesting>(
+      qec_, std::vector<IdTable>{}, vars("?x"), true,
+      std::vector<ColumnIndex>{0});
+  GroupBy groupBy{qec_, {V{"?x"}}, {}, std::move(subtree)};
+
+  expectReturningIdTables<0>(groupBy, {});
+}
+
+// _____________________________________________________________________________
+TEST_P(GroupByLazyFixture, testGeneratorWithEmptyTablesYieldsEmptyResult) {
+  std::vector<IdTable> idTables;
+  // Arbitrary count, could be anything greater than 0
+  for (size_t i = 0; i < 7; ++i) {
+    idTables.emplace_back(1, qec_->getAllocator());
+  }
+
+  auto subtree = makeExecutionTree<ValuesForTesting>(
+      qec_, std::move(idTables), vars("?x"), true, std::vector<ColumnIndex>{0});
+  GroupBy groupBy{qec_, {V{"?x"}}, {}, std::move(subtree)};
+
+  expectReturningIdTables<0>(groupBy, {});
+}
+
+// _____________________________________________________________________________
+TEST_P(GroupByLazyFixture, verifyEmptyIdTableIsSkipped) {
+  // SELECT (SUM(?x) as ?sum) WHERE { ... } GROUP BY ?y
+  std::vector<IdTable> idTables;
+  idTables.emplace_back(2, qec_->getAllocator());
+  idTables.push_back(makeIdTableFromVector({{1, 0}}, IntId));
+  idTables.emplace_back(2, qec_->getAllocator());
+  idTables.emplace_back(2, qec_->getAllocator());
+  idTables.push_back(makeIdTableFromVector({{2, 0}}, IntId));
+  idTables.emplace_back(2, qec_->getAllocator());
+  idTables.push_back(makeIdTableFromVector({{4, 1}}, IntId));
+  idTables.emplace_back(2, qec_->getAllocator());
+
+  auto subtree = makeExecutionTree<ValuesForTesting>(
+      qec_, std::move(idTables), vars("?x", "?y"), true,
+      std::vector<ColumnIndex>{1});
+  Alias alias{SparqlExpressionPimpl{makeSum("?x"), "SUM(?x)"}, V{"?sum"}};
+  GroupBy groupBy{qec_, {V{"?y"}}, {std::move(alias)}, std::move(subtree)};
+
+  expectReturningIdTables<2>(groupBy, {makeIdTableFromVector({{0, 3}}, IntId),
+                                       makeIdTableFromVector({{1, 4}}, IntId)});
+}
+
+// _____________________________________________________________________________
+TEST_P(GroupByLazyFixture, verifyImplicitGroupByWorksWithNonEmptyInput) {
+  std::vector<IdTable> idTables;
+  idTables.push_back(makeIdTableFromVector({{1}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{2}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{4}}, IntId));
```

Comment:
Consider writing a helper function (soemwhere at the top of this file) s.t. you can write
`makeIntTable({{1}})` 
and consistantly use it.